crates/api_crud/src/site/update.rs

```diff
@@ -64,6 +65,19 @@ impl PerformCrud for EditSite {
       return Err(LemmyError::from_message("application_question_required"));
     }
 
+    if let Some(default_post_listing_type) = &data.default_post_listing_type {
+      // only allow all or local as default listing types
+      let ok = match ListingType::from_str(default_post_listing_type) {
```

Comment:
Use another if let else here, much cleaner. IE `if let Ok(default_post...) = ListingType::from_str(... else { return Err`

Reply:
Did it differently (much shorter).